sdk/core/azure-core/src/main/java/com/azure/core/implementation/logging/DefaultLogger.java

```diff
@@ -0,0 +1,376 @@
+// Copyright (c) Microsoft Corporation. All rights reserved.
+// Licensed under the MIT License.
+
+package com.azure.core.implementation.logging;
+
+import com.azure.core.util.Configuration;
+import com.azure.core.util.logging.LogLevel;
+import java.io.PrintWriter;
+import java.io.StringWriter;
+import java.time.LocalDateTime;
+import java.time.format.DateTimeFormatter;
+import org.slf4j.helpers.FormattingTuple;
+import org.slf4j.helpers.MarkerIgnoringBase;
+import org.slf4j.helpers.MessageFormatter;
+
+/**
+ * This class is an internal implementation of slf4j logger.
+ */
+public final class DefaultLogger extends MarkerIgnoringBase {
+    private static final long serialVersionUID = -144261058636441630L;
+
+    private static final String AZURE_LOG_LEVEL = "AZURE_LOG_LEVEL";
+    private static final DateTimeFormatter DATE_FORMAT = DateTimeFormatter.ofPattern("yyyy-MM-dd HH:mm");
+
+    // The template forms the log message in a format:
+    // YYYY-MM-DD HH:MM [thread] [level] classpath - message
+    // E.g: 2020-01-09 12:35 [main] [WARNING] com.azure.core.DefaultLogger - This is my log message.
+    private static final String MESSAGE_TEMPLATE = "%s [%s] [%s] %s - %s";
+
+    private String classPath;
+
+    /**
+     * Construct DefaultLogger for the given class.
+     *
+     * @param clazz Class creating the logger.
+     */
+    public DefaultLogger(Class<?> clazz) {
+        this(clazz.getName());
+    }
+
+    /**
+     * Construct DefaultLogger for the given class name.
+     *
+     * @param className Class name creating the logger.
+     * @throws RuntimeException it is an error.
+     */
+    public DefaultLogger(String className) {
+        try {
+            this.classPath = Class.forName(className).getCanonicalName();
+        } catch (ClassNotFoundException e) {
+            this.classPath = className;
+        }
+    }
+
+    /**
+     * {@inheritDoc}
+     */
+    @Override
+    public boolean isDebugEnabled() {
+        String logLevelStr = Configuration.getGlobalConfiguration().get(AZURE_LOG_LEVEL);
+        LogLevel currentLogLevel = LogLevel.fromString(logLevelStr);
```

Comment:
Should this logic be abstracted into `LoggingUtils` or as a private static method on this class?

Reply:
Have a private helper method in class. Env log level is only needed for default logger, so it is ok to leave it here for current use.